src/Microsoft.Health.Fhir.Core/Features/Operations/Export/GetExportRequestHandler.cs

```diff
@@ -46,6 +46,10 @@ public async Task<GetExportResponse> Handle(GetExportRequest request, Cancellati
 
                 exportResponse = new GetExportResponse(HttpStatusCode.OK, jobResult);
             }
+            else if (outcome.JobRecord.Status == OperationStatus.Failed || outcome.JobRecord.Status == OperationStatus.Canceled)
+            {
+                exportResponse = new GetExportResponse(HttpStatusCode.InternalServerError, outcome.JobRecord.FailureReason);
```

Comment:
I don't think this is right. InternalServerError signals that this particular GetExport request has failed not that the job itself has failed. If the job is failed, it should still return 200 OK and the user can look at the OperationOutcome within the response to figure out what happened (or we can return some flag/status to indicate the result of the job).

Reply:
I think that's a valid point. But the spec (https://github.com/HL7/bulk-data/blob/master/spec/export/index.md#bulk-data-status-request) recommends returning an error code for the `get status` request if the export fails. The way I interpreted it was that `get status` serves as a "proxy" to the export status and hence it is okay to return an error code even though the `get status` request itself succeeded. And since we are returning 500 to the end user, I decided to use that at the controller level also.